### PR TITLE
Implement string width

### PR DESCRIPTION
**Pull request by duncanspumpkin**
The original didn't handle the move_x command but seemed silly not to add it. This was basically the same as wrap string but simplified.

### CODE REVIEW

File: src/OpenLoco/src/Drawing/SoftwareDrawingContext.cpp [hunk 1]
```diff
@@ -1378,10 +1378,82 @@ namespace OpenLoco::Drawing
...
-            regs.esi = X86Pointer(buffer);
-            call(0x00495715, regs);
-            return regs.cx;
+            auto font = *_currentFontSpriteBase;
+            uint16_t maxWidth = 0;
+
+            for (auto* ptr = buffer; *ptr != '\0'; ++ptr)
+            {
+                uint16_t lineWidth = 0;
+                for (; *ptr != '\0'; ++ptr)
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Drawing/SoftwareDrawingContext.cpp:1387]: Should we split up the inner for loop to another function, so we have a function that computes a just single line width? Then this 'outer' function can take the max of the width of all lines.

**Review by AaronVanGeffen — COMMENTED** [review]
This PR fixes a glitch in the hover status of the TitleExit window (which uses `getStringWidthNewLined`). Worth a changelog entry?

File: src/OpenLoco/src/Drawing/SoftwareDrawingContext.cpp (hunk 1 above)
Comment (duncanspumpkin) [reply src/OpenLoco/src/Drawing/SoftwareDrawingContext.cpp:1387]: Yeah that might be neater.

**Comment by duncanspumpkin** [comment]
> This PR fixes a glitch in the hover status of the TitleExit window (which uses `getStringWidthNewLined`). Worth a changelog entry?

What was the glitch?

**Comment by AaronVanGeffen** [comment]
> What was the glitch?

Hovering the 'Exit game' button would shrink it slightly, moving the text to the right. (Perhaps this doesn't/didn't happen on Windows?)

**Comment by duncanspumpkin** [comment]
> 

How odd. It still happens for me on this branch.

**Comment by AaronVanGeffen** [comment]
> How odd. It still happens for me on this branch.

Very odd. Now it's happening again for me as well. This may have something to do with relying on UB somewhere, then...

**Review by AaronVanGeffen — APPROVED** [review]
Code looks good to me. Thanks for splitting up the function; I think it  improves readability.

**Pull request merged**